### ISSUE TITLE
Run actual clipboard handling asynchronously

### ISSUE THREAD

**Issue by cdown** [cdown/clipmenu#59]
This would allow us to grab multiple clipboard selections if another happens while we are running the main block

**Comment by cdown** [comment]
Temporarily reverted in a85f8998b40626ac88dce5c5f32a8e9f538a983b

**Comment by cdown** [comment]
Other code was backed out in a010dee360498ebf37b1a508a1b03dcb39569429

**Comment by cdown** [comment]
We might be able to still do this, but store last_data and other state like last_cache_output in files which are also behind the flock.

**Comment by nick87720z** [comment]
Greetings. Out of curiosity, can PR #180 affect this by moving to approach with persistant pipeline? Latest what I pushed there still uses main loop in main shell level, but now using clipnotify in continuous mode (I made PR for clipnotify too). At least chance to miss clipboard events should be near zero.

I'm also about to try to try awk variant for main loop - this would, of course, still require shell participation to command awk part. But anyway - there's achievement: via process substitution - clipnotify works in subshell behind file descriptor, while main loop is still at main level and can react to events itself.

After last commit - 5deb5aaea330a8bcd95b63c26466013696eea1d1, I'm sure - entire chunks of code could be moved behind file desciptors. At least those, which don't need two-side communication (e.g. vacuuming loop could be done in this way, just taking file names continuously, independently of main loop).

I have working awk main loop implementation, but for now it's not so advanced. I did it with expectation of possible move to posix shell, so did not yet try process substitution (that would be really powerful).
